### PR TITLE
Contact Form: Enable undoing adding a contact form

### PR DESCRIPTION
**Pull request by monsieur-z**
See https://github.com/Automattic/jetpack/issues/35505

## Proposed changes:

Adding a Contact Form block in the Gutenberg editor cannot be undone. This PR fixes it.

This issue is due to some attributes being modified with the `setAttributes` function not after a user interaction but rather on mount inside a `useEffect` hook though I don't understand why that's a problem. 3 cases needed to be fixed:

1. The `to` and `subject` attributes. Instead of setting their default value client-side in a `useEffect` hook, we now get the values server-side, pass them to the client via the `window` object, and retrieve them when registering the block client-side.
2. The `requiredText` attribute. Actually, it's not in the list of block attributes, so instead of considering it as is, we simply set a fallback value when rendering a field label.
3. The `formTitle` attribute. This attribute is used to provide an accessible name label to a form for screen readers. Its computation is a bit computation-heavy, and I've decided to remove it completely for now instead of implementing a server-side version that could slow down the rendering of the page.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Spin up a test site with this branch
- Build the `forms` package and the Jetpack plugin if you're running this locally
- Create a new post
- Add a _Form_ block
- Undo the action and check it's been removed
- Add a _Contact Form_ block
- Undo the action and check it's been removed
- Add it back
- In the _Email Connection_ panel of the sidebar, verify that the email recipient and subject have a default value
Screenshot 2024-03-20 at 3 43 18 PM

### CODE REVIEW

File: projects/packages/forms/src/blocks/contact-form/class-contact-form-block.php [hunk 1]
```diff
@@ -166,6 +170,15 @@ public static function load_editor_scripts() {
 				'enqueue'    => true,
 			)
 		);
+
+		$data = array(
+			'defaults' => array(
+				'to'      => wp_get_current_user()->user_email,
+				'subject' => '[' . get_bloginfo( 'name' ) . ']' . ( isset( $post ) ? ' ' . $post->post_title : '' ),
```
Comment (jeherve) [inline projects/packages/forms/src/blocks/contact-form/class-contact-form-block.php]: I'd suggest escaping that data to avoid any potential issues.

**Review by jeherve — COMMENTED** [review]
This tests well; I only have a nitpick.

File: projects/packages/forms/src/blocks/contact-form/class-contact-form-block.php (hunk 1 above)
Comment (monsieur-z) [reply projects/packages/forms/src/blocks/contact-form/class-contact-form-block.php]: Good catch!

**Review by jeherve — APPROVED** [review]
This tests well for me. I only updated the PR description so it doesn't auto-close #35505 when we merge this PR, since the issue remains for the Donations block.

**Pull request merged**